src/poetry/publishing/publisher.py

```diff
@@ -57,8 +57,11 @@ def publish(
                 raise RuntimeError(f"Repository {repository_name} is not defined")
 
         if not (username and password):
-            # Check if we have a token first
-            token = self._authenticator.get_pypi_token(repository_name)
+            token = None
+            if not (username or password):
```

Comment:
I don't think this is desired.

Reply:
The original behavior would overwrite any value passed via CLI, if either is missing and a token is found (e.g if a username is passed and a token is found, then it would discard the username). Do we want to consider token auth if there's a username? I am happy to change back to the original behavior, although I find it a bit unintuitive.